watertap/flowsheets/full_water_resource_recovery_facility/BSM2_P_extension.py

```diff
@@ -532,12 +536,56 @@ def scale_variables(m):
         )
         iscale.set_scaling_factor(block.control_volume.material_balances, 1e3)
 
-    # Apply scaling
-    scale_variables(m)
-    iscale.calculate_scaling_factors(m)
+    # This helps to solve AD initialization to an optimal solution
+    iscale.set_scaling_factor(m.fs.AD.liquid_phase.reactions[0.0].reaction_rate, 1e5)
+    iscale.set_scaling_factor(m.fs.AD.liquid_phase.rate_reaction_generation, 1e2)
+    iscale.set_scaling_factor(m.fs.AD.liquid_phase.mass_transfer_term, 1e2)
+    iscale.set_scaling_factor(m.fs.AD.liquid_phase.rate_reaction_extent, 1e1)
+
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["S_su"], 1
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["S_fa"], 1e-1
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["S_ch4"], 1e9
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["S_IC"], 1
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["S_IN"], 1
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["S_IP"], 1
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["X_su"], 1e9
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["X_fa"], 1e9
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["X_c4"], 1e9
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["X_pro"], 1e9
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["X_ac"], 1e9
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["X_I"], 1e-1
+    )
+    iscale.set_scaling_factor(
+        m.fs.translator_asm2d_adm1.properties_out[0].conc_mass_comp["X_PHA"], 1
+    )
+
```

Comment:
Not so sure how you get these scaling factors but the `scale_variable` function is not used in this code anymore

Reply:
Well I've changed the name from `scale_variable` to `set_scaling` to be a bit more consistent with other function names, but it is being used in the code. `main` calls it after setting the operating conditions